ui/src/AudioRecorder.js

```diff
@@ -18,54 +18,65 @@ limitations under the License. */
  */
 export class AudioRecorder {
     
-    /**
-     * Create an AudioRecorder that can record utterances.
-     */
+   /**
+    * Creates an AudioRecorder that can record utterances
+    * @param {Object} allows access to Web Audio API
+    * @param {Object} Stores the recorded stream of the utterance 
+    */
     constructor() {
-        /**
-         * Allows access to Web Audio API.
-         * @private
-         */
-        this.mediaRecorder_;
+        this.mediaRecorder = undefined;
+        this.stream = undefined;
+    }
+    
+   /**
+    * Prompts user for access to Microphone using API
+    */
+    initRecorder() {
+      return new Promise(async (resolve, reject) => {
+          try {
+            this.stream = await navigator.mediaDevices.getUserMedia({ audio: true, video: false });
+            this.mediaRecorder = new MediaRecorder(this.stream);
+          } catch(err) {
+            console.log(err.name, err.message); 
+            window.alert(`Error: Microphone access is currently blocked for this site. 
+              To unblock, please navigate to chrome://settings/content/microphone and 
+              remove this site from the 'Block' section.`);
+            reject();
```

Comment:
If you reject here an error will be thrown, so that when you await initRecorder (line 52 of record button), you will also have to handle it in some way.

```
myFunction() {
await initRecorder(); // if reject(), then error is thrown and app crashes
}
```

There's two ways to handle it - with try-catch or a .catch
```
try { 
await initRecorder() 
} catch(e) { 
// handle e here
}
```
or
```
await initRecorder().catch((e) =>
// handle e here 
);
```
https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/await